scipy/stats/_continuous_distns.py

```diff
@@ -2943,6 +2943,202 @@ def _entropy(self, c):
+       on Hyperbolae", Scandinavian Journal of Statistics, Vol. 5(3),
+       pp. 151-157, 1978. https://www.jstor.org/stable/4615705
+
+    .. [2] Eberlein E., Prause K. (2002) The Generalized Hyperbolic Model:
+        Financial Derivatives and Risk Measures. In: Geman H., Madan D.,
+        Pliska S.R., Vorst T. (eds) Mathematical Finance - Bachelier
+        Congress 2000. Springer Finance. Springer, Berlin, Heidelberg.
+        https://doi.org/10.1007/978-3-662-12429-1_12
+
+    .. [3] Scott, David J, Würtz, Diethelm, Dong, Christine and Tran,
+       Thanh Tam, (2009), Moments of the generalized hyperbolic
+       distribution, MPRA Paper, University Library of Munich, Germany,
+       https://EconPapers.repec.org/RePEc:pra:mprapa:19081.
+
+    .. [4] E. Eberlein and E. A. von Hammerstein. Generalized hyperbolic
+       and inverse Gaussian distributions: Limiting cases and approximation
+       of processes. FDM Preprint 80, April 2003. University of Freiburg.
+       https://freidok.uni-freiburg.de/fedora/objects/freidok:7974/datastreams/FILE1/content
+
+    %(example)s
+
+    """
+
+    def _argcheck(self, p, a, b):
+
+        return (np.logical_and(np.abs(b) < a, p >= 0)
+                | np.logical_and(np.abs(b) <= a, p < 0))
+
+    def _logpdf(self, x, p, a, b):
+        # kve instead of kv works better for large values of p
+        # and smaller values of sqrt(a^2  - b^2)
+        @np.vectorize
+        def _logpdf_single(x, p, a, b):
+            return _stats.genhyperbolic_logpdf(x, p, a, b)
+
+        return _logpdf_single(x, p, a, b)
+
+    def _pdf(self, x, p, a, b):
+        # https://www.jstor.org/stable/4615705
+        # https://doi.org/10.1007/978-3-662-12429-1_12
+        # kve instead of kv works better for large values of p
+        # and smaller values of sqrt(a^2  - b^2)
+        @np.vectorize
+        def _pdf_single(x, p, a, b):
+            return _stats.genhyperbolic_pdf(x, p, a, b)
+
+        return _pdf_single(x, p, a, b)
+
+    def _cdf(self, x, p, a, b):
+
+        @np.vectorize
+        def _cdf_single(x, p, a, b):
+            user_data = np.array(
+                [p, a, b], float
+                ).ctypes.data_as(ctypes.c_void_p)
+            llc = LowLevelCallable.from_cython(
+                _stats, '_genhyperbolic_pdf', user_data
+                )
+
+            t1 = integrate.quad(llc, -np.inf, x)[0]
+
+            if np.isnan(t1):
+                msg = ("Infinite values encountered in scipy.special.kve. "
+                       "Values replaced by NaN to avoid incorrect results.")
+                warnings.warn(msg, RuntimeWarning)
+
+            return t1
+
+        return _cdf_single(x, p, a, b)
+
+    def _rvs(self, p, a, b, size=None, random_state=None):
+        # note: X = b * V + sqrt(V) * X  has a
+        # generalized hyperbolic distribution
+        # if X is standard normal and V is
+        # geninvgauss(p = p, b = t2, loc = loc, scale = t3)
```

Comment:
I'm not really familiar with mixtures, and it looks a bit different in, e.g. [Scott](https://mpra.ub.uni-muenchen.de/19081/1/MPRA_paper_19081.pdf) Equation 8. Is there a reference where it is written like this?

Reply:
Notation comes from [here](https://github.com/scipy/scipy/pull/13298#issuecomment-751704192), which in turn comes from the docstring of `norminversegauss` _rvs() . For a more general view, see https://en.wikipedia.org/wiki/Normal_variance-mean_mixture .

Reply:
OK, maybe I was understanding the notation; maybe the question is where are the parameters passed into generalized inverse gauss coming from? For instance, if the Scott Equation 8 formula is equivalent, why does it look so different?
Would it be obvious to someone who knows what they are doing? If not, consider adding a reference and/or other comments.
Or if @chrisb83 thinks this fine as-is, that OK, too. I believe it's correct, as I tested it quite a bit, it's just not immediately obvious why when I compare Scott Equation 8 against this.

Reply:
I think it is just a matter of juggling bw different parameterizations. I am not sure there is a reference for that one exactly.
If you replace values from Scott eq. (`\xsi = \delta**2`, `phi= \alpha**2 - beta**2`) into wiki's [alternative parameterization ](https://en.wikipedia.org/wiki/Generalized_inverse_Gaussian_distribution#Alternative_parametrization)of the gig, (which is _almost_ a loc-scale parametrization), you should get `t2` and `t3` as per my docstring i.e. `\omega = (\xsi * \phi)**0.5 = \delta * (alpha**2 - beta**2) ** 0.5` and `\nu = (\xsi / \phi)**(-0.5)  = \delta * (alpha**2 - beta**2) ** (-0.5)`

Reply:
I have updated the `rvs` test here https://github.com/scipy/scipy/pull/13298/commits/54c2ed8c1f905f4658b85822d66430f3e04a6fbc so it is aligned with the parametrization used for the other distribution tests in `TestGenHyperbolic`. I think it helps to add more comfort around the correctness of the `_rvs` as it is using different parameters than the default `test_cont_basic`, including non-trivial params for `loc` and `scale`.